corehq/apps/api/tests.py

```diff
@@ -125,6 +126,56 @@ def single_endpoint(self, id):
                                                           resource_name=self.resource.Meta.resource_name,
                                                           pk=id))
 
+    def _api_url(self, url, username=None):
+        if 'api_key' in url:
```

Comment:
This could be a bit more robust (parse the url and check that `api_key` is a param), but "api_key" isn't that likely to show up elsewhere in the url, so up to you if you want to change it.